extensions/lifecycle/LifecycleQueuePopulator.js

```diff
@@ -185,19 +222,21 @@ class LifecycleQueuePopulator extends QueuePopulatorExtension {
             }
             this.log.trace('publishing bucket replication entry', { bucket: entry.bucket });
 
-            const topic = `cold-restore-req-${locationName}`;
-            this.publish(
-                topic,
-                `${entry.bucket}/${entry.key}`,
-                JSON.stringify({
-                    bucketName: entry.bucket,
-                    objectKey: entry.key,
-                    objectVersion: value.versionId,
-                    archiveInfo: value.archive.archiveInfo,
-                    requestId: uuid(),
-                    accountId
-                }),
-            );
+            const topic = coldStorageRequest + locationName;
+            const key = `${entry.bucket}/${entry.key}`;
+            const message = JSON.stringify({
+                bucketName: entry.bucket,
+                objectKey: entry.key,
+                objectVersion: value.versionId,
+                archiveInfo: value.archive.archiveInfo,
+                requestId: uuid(),
+                accountId
+            });
+
+            this._setupProducer(topic, producer => {
+                const kafkaEntry = { key: encodeURIComponent(key), message };
```

Comment:
Why the `encodeURIComponent`? The rest of the codebase doesn't seem to be escaping the bucket/object key.

Reply:
I grabbed it from here: https://github.com/scality/backbeat/blob/development/8.4/lib/queuePopulator/QueuePopulatorExtension.js#L83, not sure it is required but since I was doing a publish manually I thought I would copy publish in this way.